pennylane_lightning/lightning_kokkos/_measurements.py

```diff
@@ -0,0 +1,481 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class implementation for state vector measurements.
+"""
+
+# pylint: disable=import-error, no-name-in-module, ungrouped-imports
+try:
+    from pennylane_lightning.lightning_kokkos_ops import MeasurementsC64, MeasurementsC128
+except ImportError:
+    pass  # HELP: Should be a complaint when kokkos_ops module is not available.
+
+from typing import Callable, List, Union
+
+import numpy as np
+import pennylane as qml
+from pennylane.devices.qubit.sampling import _group_measurements
+from pennylane.measurements import (
+    ClassicalShadowMP,
+    CountsMP,
+    ExpectationMP,
+    MeasurementProcess,
+    ProbabilityMP,
+    SampleMeasurement,
+    ShadowExpvalMP,
+    Shots,
+    StateMeasurement,
+    VarianceMP,
+)
+from pennylane.ops import Hamiltonian, SparseHamiltonian, Sum
+from pennylane.tape import QuantumScript
+from pennylane.typing import Result, TensorLike
+from pennylane.wires import Wires
+
+from pennylane_lightning.core._serialize import QuantumScriptSerializer
+
+
+class LightningMeasurements:
```

Comment:
We should ideally re-use the logic has already implemented for LQ. This can be done in a followup PR before merging to master. let's discuss offline if it's actually possible :)

My concern is with the naming convention here and in the next class! I'd suggest to explicitly use LightningKokkos in the name of classes. For example, update `LightningMeasurements` to `LightningKokkosMeasurements`.

Reply:
We will add `Kokkos` to all class names to differentiate it from LQ. However, the plan for the future is to merge all similar classes